txn_writer_test.go

```diff
@@ -522,3 +524,17 @@ func TestSafeWriter_DiffLocks(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestHasDotGit(t *testing.T) {
```

Comment:
Let's add one more test that populates `vendor/.git`, and then writes out vendor using `SafeWriter`, verifying that this feature is plumbed in properly. You can look at https://github.com/golang/dep/blob/master/txn_writer_test.go#L215 for an example of how to call the safewriter and force vendor to be rewritten.